tcl/cast.tcl

```diff
@@ -655,4 +663,24 @@ namespace eval qc::cast {
         
         return [list $from_date $to_date]
     }
+
+    proc url {string} {
+        #| Cast the given string to an url
+        set lower [string tolower $string]
+        if { [qc::is url $lower] } {
+            return $lower
+        }
+        if { [qc::is url "http://${lower}"] } {
+            return "http://${lower}"
+        }
+        return -code error -errorcode CAST "Could not cast $string to an url."
+    }
+
+    proc relative_url {string} {
+        #| Cast the given string to a relative url
+        set lower [string tolower $string]
+        if { [qc::is url -relative $lower] } {
+            return $lower
+        }        
+    }
```

Comment:
Should this proc return a `CAST` error if the string isn't a relative url? Otherwise all strings will be appear to be castable to relative urls with `qc::castable`.